deploy/lspd-install.sh

```diff
@@ -0,0 +1,257 @@
+source /etc/bash.bashrc
+
+
+# Install rust
+curl https://sh.rustup.rs -sSf | sh -s -- -y
+
+# Install bitcoin
+wget https://bitcoincore.org/bin/bitcoin-core-25.0/bitcoin-25.0-x86_64-linux-gnu.tar.gz -O /opt/bitcoin.tar.gz
+tar -xzf /opt/bitcoin.tar.gz -C /opt/
+cd /opt/bitcoin-*/bin
+chmod 710 /etc/bitcoin
+sudo install -m 0755 -t /usr/local/bin *
+
+
+cat <<EOL | sudo tee /etc/systemd/system/bitcoind.service
+[Unit]
+Description=Bitcoin daemon
+After=network.target
+[Service]
+WorkingDirectory=/var/lib/bitcoind
+ExecStart=bitcoind -daemon -pid=/run/bitcoind/bitcoind.pid -conf=/etc/bitcoin/bitcoin.conf  -datadir=/var/lib/bitcoind -startupnotify='systemd-notify --ready' -shutdownnotify='systemd-notify --stopping'
+PermissionsStartOnly=true
+ExecStartPre=/bin/chgrp bitcoin /var/lib/bitcoind
+Type=notify
+NotifyAccess=all
+PIDFile=/run/bitcoind/bitcoind.pid
+Restart=on-failure
+TimeoutStartSec=infinity
+TimeoutStopSec=600
+User=bitcoin
+Group=bitcoin
+RuntimeDirectory=bitcoind
+RuntimeDirectoryMode=0710
+ConfigurationDirectory=bitcoin
+StateDirectory=bitcoind
+StateDirectoryMode=0710
+PrivateTmp=true
+ProtectSystem=full
+ProtectHome=true
+NoNewPrivileges=true
+PrivateDevices=true
+MemoryDenyWriteExecute=true
+[Install]
+WantedBy=multi-user.target
+EOL
+
+# cat to a bitcoin.conf file
+RPCPASSWORD=$(</dev/urandom tr -dc 'A-Za-z0-9' | head -c 20)
+echo "### Bitcoin Configuration ###" >> "$CREDENTIALS"
+echo "rpcuser: lnd" >> "$CREDENTIALS"
+echo "rpcpassword: $RPCPASSWORD" >> "$CREDENTIALS"
+sudo mkdir  /etc/bitcoin/
+sudo touch /etc/bitcoin/bitcoin.conf
+cat <<EOL | sudo tee /etc/bitcoin/bitcoin.conf
+txindex=1
+daemon=1
+rpcuser=lnd
+rpcpassword=$RPCPASSWORD
+minrelaytxfee=0.00000000
+incrementalrelayfee=0.00000010
+zmqpubrawblock=tcp://127.0.0.1:28332
+zmqpubrawtx=tcp://127.0.0.1:28333
+EOL
+chmod 710  /etc/bitcoin
+sudo mkdir /home/lightning/.bitcoin/
+sudo mkdir /root/.bitcoin/
+sudo ln -s /etc/bitcoin/bitcoin.conf /home/lightning/.bitcoin/bitcoin.conf
+sudo ln -s /etc/bitcoin/bitcoin.conf /root/.bitcoin/bitcoin.conf
+###################################
+######## Install lightning ########
+###################################
+sudo mkdir /home/lightning/.lightning/
+cat <<EOL | sudo tee /home/lightning/.lightning/config
+bitcoin-rpcuser=lnd
+bitcoin-rpcpassword=$RPCPASSWORD
+bitcoin-rpcconnect=127.0.0.1
+bitcoin-rpcport=8332
+addr=:9735
+bitcoin-retry-timeout=3600
+alias="${LSPName}"
+wallet=postgres://lightning:$LIGHTNING_DB_PASSWORD@localhost:5432/lightning
+
+EOL
+git clone https://github.com/ElementsProject/lightning.git /opt/lightning
+cd /opt/lightning
+git checkout v23.05
+./configure --enable-developer
+make
+make install
+cat <<EOL | sudo tee /etc/systemd/system/lightningd.service
+[Unit]
+Description=Lightning Network Daemon (lightningd)
+After=network.target
+[Service]
+ExecStart=/usr/local/bin/lightningd --plugin=/home/lightning/.lightning/plugins/lspd_plugin  --lsp-listen=127.0.0.1:12312 --max-concurrent-htlcs=30
```

Comment:
This is missing `--dev-allowdustreserve=true` and `--allow-deprecated-apis=true`